react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.stories.tsx

```diff
@@ -0,0 +1,150 @@
+import { makeStyles } from "@material-ui/core";
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+import React from "react";
+import DeckGLMap from "../../DeckGLMap";
+
+export default {
+    component: DeckGLMap,
+    title: "DeckGLMap / Lasso Layer",
+} as ComponentMeta<typeof DeckGLMap>;
+
+const useStyles = makeStyles({
+    main: {
+        height: 500,
+        border: "1px solid black",
+        position: "relative",
+    },
+    legend: {
+        width: 100,
+        position: "absolute",
+        top: "0",
+        right: "0",
+    },
+});
+
+export const lassoLayerTemplate: ComponentStory<typeof DeckGLMap> = (args) => {
+    const [editedData, setEditedData] = React.useState(args.editedData);
+    const [enableLassoVisible, setEnableLassoVisible] =
+        React.useState<boolean>(false);
+    const handleChange1 = () => {
+        setEnableLassoVisible(true);
+    };
+
+    const handleChange2 = () => {
+        setEnableLassoVisible(false);
+    };
+    React.useEffect(() => {
+        setEditedData(args.editedData);
+    }, [args.editedData]);
+
+    return (
+        <>
+            <div className={useStyles().main}>
+                <DeckGLMap
+                    {...args}
+                    editedData={editedData}
+                    setProps={(updatedProps) => {
+                        setEditedData(updatedProps);
+                    }}
+                    enableLassoVisible={enableLassoVisible}
+                />
+            </div>
+            <button onClick={handleChange1}> Enable Lasso Selection </button>
```

Comment:
I wonder if the `enableLassoVisible` prop for the `DeckGLMap` is necessary. Can we just add the layer when this button is clicked, and remove vice versa?

Reply:
this might not be possible since it means we want to let the user to add the lasso layer by themselves. I think we should add the lassolayer into the resource data but just let the user to have the availability to control to turn it visible or not.